### PR TITLE
recursor: strip dnssec records on cache hit

### PR DESCRIPTION
**Pull request by japaric**
this is triggered by a DO=1 query followed by the almost exact same query but with DO=0

this is best reviewed on a commit by commit basis

### CODE REVIEW

File: conformance/packages/conformance-tests/src/resolver/dnssec/rfc4035/section_3/section_3_2.rs
Comment (japaric) [inline conformance/packages/conformance-tests/src/resolver/dnssec/rfc4035/section_3/section_3_2.rs:58]: we should also test DO=0 followed by DO=1. given that the resolver must always set DO=1 in its outgoing queries to nameservers then that sequence of queries should result in the second one being a cache hit. that test should go under section_4_5 so I'll add it after #2244 lands

File: crates/recursor/src/recursor.rs
Comment (djc) [inline crates/recursor/src/recursor.rs:561]: Let's invert the condition and do an early return here?

**Review by djc — APPROVED** [review]

File: crates/recursor/src/recursor.rs
Comment (japaric) [reply crates/recursor/src/recursor.rs]: done

**Comment by japaric** [comment]
can we merge this? my next PR (for #2234) would be cleaner with this change in place

**Comment by djc** [comment]
I'd like to get a review from @bluejekyll in. You can use this branch as the base branch for your next one?

**Comment by bluejekyll** [comment]
Thanks for the PR!

**Pull request merged**
